### PR TITLE
[CALCITE-3724] Presto dialect implementation

### PR DESCRIPTION
**Pull request by XuQianJin-Stars**
As illustrated in [CALCITE-3724](https://issues.apache.org/jira/browse/CALCITE-3724)

### CODE REVIEW

**Comment by XuQianJin-Stars** [comment]
> Thanks for the PR, can you explain why we need a Presto dialect ? Which unparse logic do you want to customize ?

hi @danny0405  The previous discussion is here [1]
[1] https://mail-archives.apache.org/mod_mbox/calcite-dev/202001.mbox/%3CCAHf3ZbaRZckq_v%3DSNYcBbC1D4UoZrSsxA9Q0_vpQcpXasvOoEg%40mail.gmail.com%3E

**Comment by danny0405** [comment]
> > Thanks for the PR, can you explain why we need a Presto dialect ? Which unparse logic do you want to customize ?
> 
> hi @danny0405 The previous discussion is here [1]
> [1] https://mail-archives.apache.org/mod_mbox/calcite-dev/202001.mbox/%3cCAPSgeETHa88DN7Bwqd+qez-i1=3XDa2LN3+wQdk7x_QfEXm8Ew@mail.gmail.com%3e

Thanks, you can past the conclusion/the background to the JIRA issue or the link is okey.

**Comment by DonnyZone** [comment]
Hi, @XuQianJin-Stars, could you please rebase the PR? It will be nice if you can provide presto's doc and highlight the unparse logic in Jira/PR's description.

**Comment by XuQianJin-Stars** [comment]
> Hi, @XuQianJin-Stars, could you please rebase the PR? It will be nice if you can provide presto's doc and highlight the unparse logic in Jira/PR's description.

well, but I am a bit busy lately, I will take time to deal with this later.

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -4651,7 +4707,7 @@ private void checkLiteral2(String expression, String expected) {
     sql(query).ok(expected);
   }
 
-  @Test void testCubeInSpark() {
+  @Test public void testCubeWithGroupBy() {
```
Comment (DonnyZone) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:4710]: I remember that there was a PR that removed `public`. We'd better follow this style.

Comment (XuQianJin-Stars) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:4710]: > I remember that there was a PR that removed `public`. We'd better follow this style.

well, let me change it.

**Comment by DonnyZone** [comment]
LGTM

**Pull request closed without merging**